dashboard/modules/job/tests/test_job_manager.py

```diff
@@ -576,5 +581,28 @@ def test_logs_streaming(job_manager):
     job_manager.stop_job(job_id)
 
 
+def test_boostrap_address(job_manager, monkeypatch):
+    """Ensure we always use boostrap address in job manager  even though ray
+    cluster might be started with http://ip:{dashboard_port} from previous
+    runs.
+    """
+    ip = ray.ray_constants.DEFAULT_DASHBOARD_IP
+    port = ray.ray_constants.DEFAULT_DASHBOARD_PORT
+
+    monkeypatch.setenv("RAY_ADDRESS", f"http://{ip}:{port}")
+    print_ray_address_cmd = ("python -c\""
+                             "import os;"
+                             "import ray;"
+                             "ray.init();"
+                             "print('SUCCESS!');"
+                             "\"")
+
+    job_id = job_manager.submit_job(entrypoint=print_ray_address_cmd)
+
+    wait_for_condition(
+        check_job_succeeded, job_manager=job_manager, job_id=job_id)
+    assert "SUCCESS!" in job_manager.get_job_logs(job_id)
+
```

Comment:
To reproduce the bug conditions, do we need to restart the Ray cluster in the test at some point?

Reply:
not really -- all needed is to reproduce the issue is to run an entrypoint script with ray.init() that is submitted with RAY_ADDRESS="http://...", so this test just need to set it that way and ensure job submission will still finish.